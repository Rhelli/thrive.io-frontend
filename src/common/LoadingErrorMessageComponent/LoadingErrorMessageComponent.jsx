import React from 'react';
import styles from './LoadingErrorMessageComponent.module.scss';

const LoadingErrorMessageComponent = message => (
  <div className={styles.loadingErrorContainer}>
    <h2>Oops!</h2>
    <div>
      <p>
        Error!
        {' '}
        {message.message}
      </p>
    </div>
  </div>
);

export default LoadingErrorMessageComponent;
