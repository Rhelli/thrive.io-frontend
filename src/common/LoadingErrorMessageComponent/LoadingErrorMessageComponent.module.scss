.loadingErrorContainer {
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 60%;
  transform: translateY(-60%);
  background-color: #fef1ea;
  background-image: url('../../assets/img/inspiration-geometry.png');
  border: 4px solid #f69a69;
  border-radius: 50px;
  width: 60%;
  padding: 20px;
  -webkit-box-shadow: 0 0 5px 1px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 0 0 5px 1px rgba(0, 0, 0, 0.75);
  box-shadow: 0 0 5px 1px rgba(0, 0, 0, 0.75);

  h2 {
    font-size: 28px;
    text-decoration: underline;
    margin-bottom: 20px;
  }

  div {
    p {
      font-size: 16px;
    }
  }
}
