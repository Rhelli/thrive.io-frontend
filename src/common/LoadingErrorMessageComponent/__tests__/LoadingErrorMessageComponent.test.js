import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import LoadingErrorMessageComponent from '../LoadingErrorMessageComponent';

const message = {
  message: 'this is the message',
};

const loadingErrorMessageComponent = (
  <LoadingErrorMessageComponent message={message} />
);

describe('The appearance and functionality of the LoadingErrorMessage component', () => {
  it('Matches the LoadingErrorMessage snapshot', () => {
    const tree = renderer.create(loadingErrorMessageComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Contains a master container with the correct styles class', () => {
    const { container } = render(loadingErrorMessageComponent);
    expect(container.firstChild.nodeName).toBe('DIV');
    expect(container.firstChild.classList.contains('loadingErrorContainer')).toBe(true);
  });

  it('Contains and displays the error message title', () => {
    const { container, getByText } = render(loadingErrorMessageComponent);
    expect(container.firstChild.firstChild.nodeName).toBe('H2');
    getByText('Oops!');
  });

  it('Contains and displays the error message provided by props in a paragraph tag', () => {
    const { container, getByText } = render(loadingErrorMessageComponent);
    expect(container.firstChild.lastChild.firstChild.nodeName).toBe('P');
    getByText('Error! this is the message');
  });
});
