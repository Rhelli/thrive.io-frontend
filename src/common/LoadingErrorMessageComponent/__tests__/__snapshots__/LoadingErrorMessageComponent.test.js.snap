// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the LoadingErrorMessage component Matches the LoadingErrorMessage snapshot 1`] = `
<div
  className="loadingErrorContainer"
>
  <h2>
    Oops!
  </h2>
  <div>
    <p>
      Error!
       
      this is the message
    </p>
  </div>
</div>
`;
